### PR TITLE
HCO BlogPost

### PR DESCRIPTION
**Pull request by DirectedSoul1**
Please give me some comments which can improve the BlogPost for HCO

### CODE REVIEW

File: _posts/2019-04-17-Hyper-Converged-Operator.markdown [hunk 1]
```diff
@@ -147,10 +162,10 @@ virt-handler-hqz9d                                 1/1     Running   0
 virt-operator-667b6c845d-jfnsr                     1/1     Running   0          11m
 ```
 
-We can see the CRD's in the `kubevirt-hyperconverged` NameSpace
+We can see the CRD's in the which has a cluster-wide scope:
```
Comment (iranzo) [inline _posts/2019-04-17-Hyper-Converged-Operator.markdown]: We can see the CRD's with a cluster-wide scope: 

Instead?

File: _posts/2019-04-17-Hyper-Converged-Operator.markdown [hunk 2]
```diff
@@ -182,10 +197,43 @@ virt-operator                     1/1     1            1           16m
 ```
 #> **NOTE**: Here, Once we applied the Custom Resource the operator took care of deploying the actual KubeVirt pods (virt-api, virt-controller and virt-handler), CDI pods(cdi-upload-proxy, cdi-apiserver, cdi-deployment, cdi-operator) and Network add-on pods ( cluster-network-addons-operator)  . Again we’ll need to execute the command until everything is up&running (or use -w).
 
+## Deploying HCO Openshift4 Cluster.
```
Comment (iranzo) [inline _posts/2019-04-17-Hyper-Converged-Operator.markdown]: Minikube setup is level 3 (###)

File: _posts/2019-04-17-Hyper-Converged-Operator.markdown [hunk 3]
```diff
@@ -251,50 +299,50 @@ spec:
 So, after HCO is up and running we need to test it by deploying a small instance of a VM.For doing so, we need to follow the steps:
 
 ```
-export KUBEVIRT_VERSION="v0.15.0"
+$export KUBEVIRT_VERSION="v0.16.1"
```
Comment (iranzo) [inline _posts/2019-04-17-Hyper-Converged-Operator.markdown]: Is there a way to get the variable from a shared file and have all the files picking it up from there?

Also applies to minikube tutorial, etc

File: _posts/2019-04-17-Hyper-Converged-Operator.markdown [hunk 4]
```diff
@@ -251,50 +299,50 @@ spec:
 So, after HCO is up and running we need to test it by deploying a small instance of a VM.For doing so, we need to follow the steps:
```
Comment (iranzo) [inline _posts/2019-04-17-Hyper-Converged-Operator.markdown]: Should we reuse from what we already have in LAB1 ?  creates vm, starts, uses virctl, etc ?

File: _posts/2019-04-17-Hyper-Converged-Operator.markdown [hunk 5]
```diff
@@ -251,50 +299,50 @@ spec:
...
+$kubectl get vmis -o yaml testvm
 ```
 #**Note**: The 'i' in **vmi** , as it stands for VirtualMachineInstance. Now, pay attention to the phase field, its value will be transitioning from one state to the next, indicating VMI progress to finish being set to Running.
 
 
 Now use **virtctl** command to connect to the VMI consoles interfaces:
 ```
-./virtctl console HCO-vm
-./virtctl vnc HCO-vm
+$./virtctl console testvm
```
Comment (iranzo) [inline _posts/2019-04-17-Hyper-Converged-Operator.markdown]: reuse code snippets from actual labs? (check lab1)

File: _posts/2019-04-17-Hyper-Converged-Operator.markdown [hunk 6]
```diff
@@ -304,17 +352,17 @@ Now its time for a Clean Up:
...
 Delete the VM:
 ```
-kubectl delete vm HCO-vm
+$kubectl delete vm testvm
 ```
 
 Delete the minikube instance:
 ```
-minikube delete -p kubevirt-hco
+$minikube delete -p kubevirt-hco
```
Comment (iranzo) [inline _posts/2019-04-17-Hyper-Converged-Operator.markdown]: I think that we could direct to the quickstart minikube tutorial as most of the steps are from there, outside of deploying the HCO, in that way we could resuse content and reduce duplicates for later maintenance

Comment (DirectedSoul1) [reply _posts/2019-04-17-Hyper-Converged-Operator.markdown]: added the reference link

File: _posts/2019-04-17-Hyper-Converged-Operator.markdown (hunk 5 above)
Comment (DirectedSoul1) [reply _posts/2019-04-17-Hyper-Converged-Operator.markdown]: same as above.

File: _posts/2019-04-17-Hyper-Converged-Operator.markdown (hunk 4 above)
Comment (DirectedSoul1) [reply _posts/2019-04-17-Hyper-Converged-Operator.markdown]: As this is a duplicate step from minikube quick start , I have added the link for the same.

File: _posts/2019-04-17-Hyper-Converged-Operator.markdown (hunk 3 above)
Comment (DirectedSoul1) [reply _posts/2019-04-17-Hyper-Converged-Operator.markdown]: Included the link for minikube quick start.

File: _posts/2019-04-17-Hyper-Converged-Operator.markdown (hunk 1 above)
Comment (DirectedSoul1) [reply _posts/2019-04-17-Hyper-Converged-Operator.markdown]: Added the cluster-wide scope for CRD's

File: _posts/2019-04-17-Hyper-Converged-Operator.markdown [hunk 7]
```diff
@@ -24,17 +24,16 @@ The goal of the hyperconverged-cluster-operator (HCO) is to provide a single ent
...
 ### Environment description
 We can use HCO both on minikube and also on Openshift4. 
 
-* **Minikube**
-
 Minikube is a tool that makes it easy to run Kubernetes locally. Minikube runs a single-node Kubernetes cluster inside a VM on your laptop for users looking to try out Kubernetes or develop with it day-to-day.
 For testing Hyper Converged Operator I have deployed a single-node K8's cluster with the name kubevirt-hco. This can be deployed using [minikube](https://kubernetes.io/docs/setup/minikube/#installation):
 
-We’ll create a profile for KubeVirt so it gets its own settings without interfering what any configuration you might had already, let’s start by increasing the default memory to 4GiB:
+We'll create a profile for KubeVirt allowing us to define specific settings and to ensure the settings don't interfering what any configuration you might had already, let’s start by increasing the default memory to 4GiB:
```
Comment (iranzo) [inline _posts/2019-04-17-Hyper-Converged-Operator.markdown]: interfering -> interfere

Comment (iranzo) [reply _posts/2019-04-17-Hyper-Converged-Operator.markdown]: what -> with (after interfere)

File: _posts/2019-04-17-Hyper-Converged-Operator.markdown [hunk 8]
```diff
@@ -294,24 +295,153 @@ spec:
...
+[KubeVirt](http://kubevirt.io/quickstart_minikube/) : KubeVirt technology addresses the needs of development teams that have adopted or want to adopt [Kubernetes](https://kubernetes.io/) but possess existing Virtual Machine-based workloads that cannot be easily containerized. More specifically, the technology provides a unified development platform where developers can build, modify, and deploy applications residing in both Application Containers as well as Virtual Machines in a common, shared environment. 
+
+# [HCO using the OLM method](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/Documentation/design/architecture.md) 
 
-So, after HCO is up and running we need to test it by deploying a small instance of a VM.For doing so, please follow the instructions given in the [minikube_quickstart](https://kubevirt.io//quickstart_minikube/#install-virtctl):
+Replace <docker_org> with your Docker organization as official operator-registry images forHCO will not be provided.
```
Comment (iranzo) [inline _posts/2019-04-17-Hyper-Converged-Operator.markdown]: space between 'for' and 'HCO'

File: _posts/2019-04-17-Hyper-Converged-Operator.markdown [hunk 9]
```diff
@@ -294,24 +295,153 @@ spec:
...
 $minikube delete -p kubevirt-hco
 ```
 
-#**Conclusion**-What to expect next ?
+## Conclusion
+
+What to expect next ?
 
-HCO achieved its goal which was to provide a single entrypoint for multiple operators - kubevirt, cdi, networking, etc.where users can deploy and configure them in a single object as seen above.
+HCO achieved its goal which was to provide a single entrypoint for multiple operators - kubevirt, cdi, networking, etc.where users can deploy and configure them in a single object asseen above.
```
Comment (iranzo) [inline _posts/2019-04-17-Hyper-Converged-Operator.markdown]: 'as seen'

File: _posts/2019-04-17-Hyper-Converged-Operator.markdown [hunk 10]
```diff
@@ -294,24 +295,153 @@ spec:
...
 
 Now, we can also launch the HCO through OLM, 
 
-#**Note**: 
+**Note**: 
 Until we publish (and consume) the HCO and component operators through Marketplace|[operatorhub.io](https://operatorhub.io/), this is a means to demonstrate the HCO workflow without OLM
```
Comment (iranzo) [inline _posts/2019-04-17-Hyper-Converged-Operator.markdown]: 'this is a mean' (removed 's')

File: _posts/2019-04-17-Hyper-Converged-Operator.markdown (hunk 7 above)
Comment (DirectedSoul1) [reply _posts/2019-04-17-Hyper-Converged-Operator.markdown]: Added the above changes.

File: _posts/2019-04-17-Hyper-Converged-Operator.markdown (hunk 8 above)
Comment (DirectedSoul1) [reply _posts/2019-04-17-Hyper-Converged-Operator.markdown]: Added the above change.

File: _posts/2019-04-17-Hyper-Converged-Operator.markdown (hunk 9 above)
Comment (DirectedSoul1) [reply _posts/2019-04-17-Hyper-Converged-Operator.markdown]: Added the `space` between words

File: _posts/2019-04-17-Hyper-Converged-Operator.markdown [hunk 11]
```diff
@@ -0,0 +1,326 @@
...
+nodemaintenances.kubevirt.io                                     2019-05-07T20:20:58Z
+virtualmachineinstancemigrations.kubevirt.io                     2019-05-07T20:23:02Z
+virtualmachineinstancepresets.kubevirt.io                        2019-05-07T20:23:01Z
+virtualmachineinstancereplicasets.kubevirt.io                    2019-05-07T20:23:02Z
+virtualmachineinstances.kubevirt.io                              2019-05-07T20:23:01Z
+virtualmachines.kubevirt.io                                      2019-05-07T20:23:02Z
+```
+#Note: In Openshift we can use both `kubectl` and  `oc` interchangeably to interact with the cluster objects once HCO is up and running.
+
+## You can read more about CDI, CNI, ssp-operator, web-ui and KubeVirt:
```
Comment (SchSeba) [inline _posts/2019-04-17-Hyper-Converged-Operator.markdown]: CNI -> CNA

File: _posts/2019-04-17-Hyper-Converged-Operator.markdown [hunk 12]
```diff
@@ -0,0 +1,326 @@
...
+
+- [CDI](https://github.com/kubevirt/kubevirt.github.io/blob/master/_posts/2018-10-10-CDI-DataVolumes.markdown) 
+
+- [CNI](https://github.com/intel/multus-cni/blob/master/doc/quickstart.md)
```
Comment (SchSeba) [inline _posts/2019-04-17-Hyper-Converged-Operator.markdown]: CNA https://github.com/kubevirt/cluster-network-addons-operator/blob/master/README.md

Comment (DirectedSoul1) [reply _posts/2019-04-17-Hyper-Converged-Operator.markdown]: Updated the links

File: _posts/2019-04-17-Hyper-Converged-Operator.markdown (hunk 11 above)
Comment (DirectedSoul1) [reply _posts/2019-04-17-Hyper-Converged-Operator.markdown]: Updated the Term

File: _posts/2019-04-17-Hyper-Converged-Operator.markdown [hunk 13]
```diff
@@ -0,0 +1,326 @@
...
+
+# [HCO known as Hyper Converged Operator](https://github.com/kubevirt/hyperconverged-cluster-operator)  
+
+**Prerequisites:** 
+ 
+This Blog assumes that the reader is aware of the concept of Operators and how it works in K8's environment. Before proceeding further, feel free to take a look at this concept using [CoreOS BlogPost](https://coreos.com/blog/introducing-operators.html)
+
+## What it does?
+
+The goal of the hyperconverged-cluster-operator (HCO) is to provide a single entrypoint for multiple operators - [kubevirt](https://blog.openshift.com/a-first-look-at-kubevirt/), [cdi](http://kubevirt.io/2018/CDI-DataVolumes.html), [networking](https://github.com/intel/multus-cni/blob/master/doc/quickstart.md), etc... - where users can deploy and configure them in a single object. This operator is sometimes referred to as a "meta operator" or an "operator for operators". Most importantly, this operator doesn't replace or interfere with OLM. It only creates operator CRs, which is the user's prerogative.
```
[the remark this answers is not in the document]
Comment (DirectedSoul1) [reply _posts/2019-04-17-Hyper-Converged-Operator.markdown]: Added the relevant links

File: _posts/2019-04-17-Hyper-Converged-Operator.markdown (hunk 12 above)
Comment (iranzo) [reply _posts/2019-04-17-Hyper-Converged-Operator.markdown]: Remove newlines between '-' so that it doesn't show as double spaced:

image

Comment (iranzo) [reply _posts/2019-04-17-Hyper-Converged-Operator.markdown]: It should be like:

- [CDI]XXXX
- [CNI]XXXX

**Comment by iranzo** [comment]
@DirectedSoul1 I've uploaded a commit from my side to this review, moving images to the assets/HCO folder and fixing the reference to them instead the absolute URI to your original repo.

Also removed some double spacing, spaces at end of line or title, etc.

I've validated on the docker image and seems to render properly here, can you check on your side too?

Thanks

**Comment by iranzo** [comment]
@DirectedSoul1 I've also fixed a link that was failing verification of CI on a secondary post.

IMHO, each PR should contain only one article unless something requires several files at the same time (like cross-references)

According to current PR branch against master, it shows several files changed:

diff --git a/HCO-minikube.markdown b/HCO-minikube.markdown
diff --git a/_posts/2019-04-17-Hyper-Converged-Operator.markdown b/_posts/2019-04-17-Hyper-Converged-Operator.markdown
diff --git a/_posts/2019-05-15-How-To-Guide-Import-VM-Kubevirt.markdown b/_posts/2019-05-15-How-To-Guide-Import-VM-Kubevirt.markdown
diff --git a/assets/HCO/HCO-Managed-Operators.png b/assets/HCO/HCO-Managed-Operators.png
diff --git a/assets/HCO/Screenshot from 2019-05-08 14-02-31.png b/assets/HCO/Screenshot from 2019-05-08 14-02-31.png
diff --git a/assets/HCO/Screenshot from 2019-05-08 14-03-57.png b/assets/HCO/Screenshot from 2019-05-08 14-03-57.png
diff --git a/assets/HCO/Screenshot from 2019-05-08 14-04-48.png b/assets/HCO/Screenshot from 2019-05-08 14-04-48.png
diff --git a/assets/HCO/Screenshot from 2019-05-08 14-10-32.png b/assets/HCO/Screenshot from 2019-05-08 14-10-32.png
diff --git a/assets/HCO/Screenshot from 2019-05-08 14-12-15.png b/assets/HCO/Screenshot from 2019-05-08 14-12-15.png
diff --git a/assets/HCO/Screenshot from 2019-05-08 14-12-54.png b/assets/HCO/Screenshot from 2019-05-08 14-12-54.png

**Comment by iranzo** [comment]
Probably, towards final blog post, it would be good to have the 'screenshoot' files renamed to a description on what they show and have the markdown reference updated accordingly

**Comment by codificat** [comment]
As this PR became quite complex, containing multiple posts and with a problematic commit history, closing it in favour of simpler and cleaner PRs. See #228 for the "import VM" blog post and #229 for the HCO blog post

**Pull request closed without merging**